Sources/ManageMaintenance.php

```diff
@@ -2291,14 +2212,38 @@ function get_hook_info_from_raw($rawData)
 	{
 		list ($hookData['class'], $hookData['method']) = explode('::', $modFunc);
 		$hookData['pureFunc'] = $hookData['method'];
+		$hookData['call'] = $modFunc;
 	}
 
 	else
-		$hookData['pureFunc'] = $modFunc;
+		$hookData['call'] = $hookData['pureFunc'] = $modFunc;
 
 	return $hookData;
 }
 
+function get_defined_functions_in_file(string $file): array
+{
+	$source = file_get_contents($file);
+	// token_get_all() is too slow so use a nice little regex instead.
+	preg_match_all('/^namespace\s([a-z\\\]++)\s*;|class\s+(\w++)[\w\s]*{|function\s+(\w++)\s*\(.*\)[:\|\w\s]*{/im', $source, $matches, PREG_SET_ORDER);
```

Comment:
```suggestion
	preg_match_all('/\bnamespace\s++((?P>label)(?:\\\(?P>label))*+)\s*+;|\bclass\s++((?P>label))[\w\s]*+{|\bfunction\s++((?P>label))\s*+\(.*\)[:\|\w\s]*+{(?(DEFINE)(?<label>[a-zA-Z_\x80-\xff][a-zA-Z0-9_\x80-\xff]*+))/i', $source, $matches, PREG_SET_ORDER);
```
1. Uses a `(?P>label)` subroutine, which is defined as `(?<label>[a-zA-Z_\x80-\xff][a-zA-Z0-9_\x80-\xff]++)` inside the the `(?(DEFINE)...)` block, to more precisely match valid identifier labels for functions/classes/namespaces. The `[a-zA-Z_\x80-\xff][a-zA-Z0-9_\x80-\xff]*` definition comes from the PHP documentation [here](https://www.php.net/manual/en/language.variables.basics.php).
2. Uses `\b` before `namespace`, `class`, and `function` to better protect against, e.g. `looks_like_a_function_but_is_not`. This also allows removing the `^` and the need for multiline mode.